platform/qt/qt.cmake

```diff
@@ -9,19 +9,25 @@ message(STATUS "Version ${MBGL_QT_VERSION}")
 
 option(MBGL_QT_STATIC "Build Mapbox GL Qt bindings staticly" OFF)
 option(MBGL_QT_LIBRARY_ONLY "Build only libraries" OFF)
+option(MBGL_WITH_QT_HEADLESS "Build Mapbox GL Qt with headless support" ON)
```

Comment:
What's the idea of this headless version? Isn't it basically the same as MBGL_QT_LIBRARY_ONLY?

Reply:
Headless requires OpenGL and widgets. Library only allows you to build a version which is running on GLES. This is important for platforms without QtWidgets or OpenGL - such as some mobile Linux distributions.

Reply:
For some reason I still don't undestand this "headless" business :). Can you provide a few simple examples for all three (none / headless / library) usage? I can think of only three:

1. Running the Maplibre Qt demo app directly from this reposotory.
2. Including Maplibre into some other build tree (as a git submodule, for example, like QtLocation plugin).
3. Building a library, making a package, and making find_package from some other applicationäs build tree.

Reply:
Headless component gets on a way if you don't have OpenGL, as some Qt platforms do. Corresponding code: https://github.com/maplibre/maplibre-gl-native/blob/master/platform/qt/src/headless_backend_qt.cpp . Hence I want to be able to drop it, as I do on Sailfish. To your examples:

1. demo app - that I haven't checked thoroughly. as it is using QtWidgets and, in Sailfish case we don't have that either, headless is enabled
2. git submodule. that would depend on what you use in submodule, tbh. I don't know what is in QtLocation plugin precisely and don't know whether it is exposing headless functionality. if you don't use it, there is no need for it and you may want to skip OpenGL requirements as well.
3. separate lib: again, as in 2, it depends on platform and your application. in my case, headless is no go and I will disable it before linking with my QML plugin. For some other users (if we have any), it could be different.

As not many know about the flag, I would suggest to keep it enabled by default.

Hopefully, I replied to you.

Reply:
Aha, you mean that you don't have OpenGL Qt module? Because you must have OpenGL in the system in general.

But, about the `headless_backend_qt.cpp` - why do we need it at all? For tests? I think I'm missing something here.

Reply:
Sorry, missed it last night. On some platforms you may not have OpenGL fully. For those, there is no Qt module either :). Same goes for QtWidgets.

I haven't used it, but I think headless could be used if you need to have a map tile rendered using MapboxGL. Could be handy, actually.